app/controllers/areas_controller.rb

```diff
@@ -18,36 +21,27 @@ def edit
 
   def create
     @area = Area.new(area_params)
-    respond_to do |format|
-      if @area.save
-        format.html { redirect_to @area, notice: 'Area was successfully created.' }
-        format.json { render :show, status: :created, location: @area }
-      else
-        format.html { render :new }
-        format.json { render json: @area.errors, status: :unprocessable_entity }
-      end
-    end
+    @area.city_id=params[:city_id]
+   if @area.save
+     @areas = @city.areas.all
```

Comment:
after creation of new area in city,  I am updating areas table which is below the area creation form(on same page). and all this things are done using ajax. so for updated view of area im fetching all areas after create action